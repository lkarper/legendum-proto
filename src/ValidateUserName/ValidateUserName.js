import React, { useEffect } from 'react';

const ValidateUserName = (props) => {
    const { userNameError } = props;

    useEffect(() => {
        const { setUserNameError, userName } = props;
        if (userName.length && !userName.trim().length) {
            setUserNameError('Username must contain at least 1 character')
        } else {
            setUserNameError(null);
        }
    }, [props]);

    if (userNameError) {
        return (
            <div role="alert">
                <p id="username-required">{userNameError}</p>
            </div>
        );
    }
    return <p id="username-required">Username required.</p>;
}

export default ValidateUserName;