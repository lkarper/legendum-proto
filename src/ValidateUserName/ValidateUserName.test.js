import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ValidateUserName from './ValidateUserName';

describe('ValidateUserName component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ValidateUserName />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<ValidateUserName />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
