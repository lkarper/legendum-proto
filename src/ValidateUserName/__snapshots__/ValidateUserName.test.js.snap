// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidateUserName component renders the UI as expected 1`] = `
<ValidateUserName
  setUserNameError={[Function]}
  userName=""
  userNameError=""
>
  <div
    role="alert"
  >
    <p
      id="username-required"
    >
      Username meets requirements.
    </p>
  </div>
</ValidateUserName>
`;
